### PR TITLE
Fixing reply count on new posts

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing reply count on new posts

#### Ticket Link
[MM-17468](https://mattermost.atlassian.net/browse/MM-17468)

### CODE REVIEW

**Comment by jespino** [comment]
@reflog sorry, I made some changes to fix another problem.

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -703,11 +703,9 @@ func (s *SqlPostStore) getPostsAround(before bool, options model.GetPostsOptions
 		direction = ">"
 		sort = "ASC"
 	}
-	replyCountSubQuery := s.getQueryBuilder().Select("COUNT(Posts.Id)").From("Posts").Where(sq.Expr("p.RootId = '' AND RootId = p.Id AND DeleteAt = 0"))
+	replyCountSubQuery := s.getQueryBuilder().Select("COUNT(Posts.Id)").From("Posts").Where(sq.Expr("Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END) AND Posts.DeleteAt = 0"))
```
Comment (reflog) [inline store/sqlstore/post_store.go:706]: shouldn't it be:
`(CASE WHEN Posts.RootId = '' THEN Posts.Id ELSE Posts.RootId END) `

Comment (jespino) [reply store/sqlstore/post_store.go:706]: I don't think so, the `p.RootId` and `p.Id` is the post from outside the subquery.

**Pull request merged**